src/pages/upload/index.tsx

```diff
@@ -28,12 +30,57 @@ const fileUploadReducer = (state: any, action: any) => {
   }
 }
 
-const Upload: React.FC = () => {
-  const [state, dispatch] = React.useReducer(fileUploadReducer, {files: []})
+type UploadedFile = {
+  fileName: string
+  signedUrl: string
+}
+
+type LessonMetadata = {
+  title: string
+  fileMetadata: UploadedFile
+}
+
+type FormProps = {
+  lessons: LessonMetadata[]
+}
+
+const UploadWrapper = () => {
   const {viewer} = useViewer()
+  const initialValues: FormProps = {lessons: []}
+
+  if (!viewer?.s3_signing_url) return null
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={async (values, actions) => {
+        const response = await axios.post('api/sanity/lessons/create', {
+          lessons: values.lessons,
+        })
+
+        console.log({response})
+      }}
+    >
+      {(props) => <Upload {...props} />}
+    </Formik>
+  )
+}
+
+const Upload: React.FC<FormikProps<FormProps>> = (formikProps) => {
+  const {values, setFieldValue} = formikProps
+
+  const {viewer} = useViewer()
+  const [state, dispatch] = React.useReducer(fileUploadReducer, {files: []})
```

Comment:
suggestion(style): consider naming `state` something more descriptive in terms of what the state IS beyond generically being "state"